src/main/java/com/capgemini/fs/coindashboard/apiCommunicator/coinGeckoCommunicator/CoinGeckoFacade.java

```diff
@@ -186,7 +191,18 @@ public Optional<Result> getCoinInfo(List<String> coins) {
       Result result = new Result(this.provider, ResultStatus.FAILURE, null, new ArrayList<>());
       List<String> errorMessages = new ArrayList<>();
 
-      for (String coin : coins) {
+      for (int c = 0; c < coins.size(); c++) {
+        String coin = coins.get(c);
+        if (c > 19) {
+          if (c == 20) {
+            log.info("{} Waiting 60 seconds to not block CoinGecko", c+1);
+            sleep(60000l);
```

Comment:
https://stackoverflow.com/questions/54394042/java-how-to-avoid-using-thread-sleep-in-a-loop